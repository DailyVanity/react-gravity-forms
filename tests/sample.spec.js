// import { render, toJSON } from "@testing-library/react";
import renderer from 'react-test-renderer';
import * as React from "react";
import GravityForm from "../src";
// jest.useFakeTimers()

it("should render the form", async () => {
  const component = renderer.create(
    <GravityForm
      backendUrl={`https://backend.glamrock.dev.gohike.nl/wp/wp-json/glamrock/v1/gf/forms`}
      formID={"1"}
      title={"titel"}
      // onChange={onChange}
      // populatedFields={searchParams}
      // styledComponents={{
      //   Label,
      //   Input,
      //   Textarea,
      //   Checkbox,
      //   Radiogroup,
      //   GFWrapper,
      //   FileWrapper,
      //   Consent,
      //   Button,
      // }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseEnter();
  // });
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // renderer.act(() => {
  //   tree.props.onMouseLeave();
  // });
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
