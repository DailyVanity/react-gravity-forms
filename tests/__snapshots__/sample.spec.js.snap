// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the form 1`] = `
<div
  className="form-wrapper"
  id="gravity_form_1"
/>
`;
